GDJS/Runtime/spriteruntimeobject.ts

```diff
@@ -845,41 +853,73 @@ namespace gdjs {
      * (x and y position of the point in global coordinates).
      */
     private _transformToGlobal(x: float, y: float, result: number[]) {
+      result.length = 2;
+      const onlyNeedTranslation =
+        this.angle === 0 &&
+        this._scaleX === 0 &&
+        this._scaleY === 0 &&
+        !this._flippedX &&
+        !this._flippedY;
+      if (onlyNeedTranslation) {
```

Comment:
We need to optimize this. The flipped check is useless.

Reply:
This is only one run, but that's better.

* getAABB of a non rotated sprite, with custom hitboxes, origin and center(60000x): 47.20833333283663,
* getAABB of a rotated sprite, with custom hitboxes, origin and center(60000x): 61.76666666641832,
* getAABB of a rotating sprite, with custom hitboxes, origin and center(60000x): 71.03166666701436